python/ray/remote_function.py

```diff
@@ -21,6 +21,32 @@
 logger = logging.getLogger(__name__)
 
 
+_task_launch_hook = None
+
+
+def tune_task_launch_hook(fn, resources):
+    if "_export_bundle" in fn.function_name:  # delete this later
+        return
+    if not resources or resources == {"CPU": 0}:
+        return
+    print("TASK", fn, resources)
+    # TODO: only raise if Tune has no extra resources configured for the trial
+    name = fn.module_name + "." + fn.function_name
+    from ray.tune.error import AbortTrialExecution
+    if resources in [{"CPU": 1}, {"CPU": 1, "GPU": 0}]:
+        raise AbortTrialExecution(
+            f"Your trial launched the Ray task `{name}`, but no extra CPUs were "
+            "reserved for task execution. Specify the Tune argument "
+            '`per_trial_task_resources={"cpu": N}` to reserve CPUs for task execution.')
```

Comment:
The user could also specify the placement group factory.

Reply:
We should settle down on whether we want PGF to be end user API. It's pretty unintuitive IMO.

Reply:
Agreed.